src/client.rs

```diff
@@ -43,6 +45,47 @@ use crate::{
     AstarteAggregate, Error, Interface,
 };
 
+type DynError = Box<dyn std::error::Error + Send + Sync + 'static>;
```

Comment:
The `DynError` is also used in the store  (and maybe other places), we can move it to the src/error.rs file and make it pub(crate)